src/lib/deployWorkers.ts

```diff
@@ -488,126 +488,131 @@ export const prepareForDeploy = async ({
+          commandObj.error(
+            `You must have at least one peerId listed in ${color.yellow(
+              "peerIds",
+            )} property in ${color.yellow(
+              `hosts.${workerName}`,
+            )} property in ${fluenceConfig.$getPath()}`,
           );
+        }
 
-          const { overrideModules, serviceConfig } = maybeServiceConfig;
+        const workerConfig = workersFromFluenceConfig[workerName];
 
-          const { [FACADE_MODULE_NAME]: facadeModule, ...restModules } =
-            serviceConfig.modules;
-
-          const modules = [
-            ...Object.entries(restModules),
-            [FACADE_MODULE_NAME, facadeModule] as const,
-          ].map(([name, { get, ...overridesFromService }]) => {
-            const moduleUrlOrAbsolutePath = getUrlOrAbsolutePath(
-              get,
-              serviceConfig.$getDirPath(),
-            );
+        assert(
+          workerConfig !== undefined,
+          `Unreachable. workerNamesNotFoundInWorkersConfig was empty but error still happened. Looking for ${workerName} in ${JSON.stringify(
+            workersFromFluenceConfig,
+          )}`,
+        );
 
-            const moduleConfig = moduleAbsolutePathOrURLToModuleConfigsMap.get(
-              moduleUrlOrAbsolutePath,
-            );
+        const services: Upload_deployArgConfig["workers"][number]["config"]["services"] =
+          (workerConfig.services ?? []).map((serviceName) => {
+            const maybeServiceConfig = serviceConfigs.find((c) => {
+              return c.serviceName === serviceName;
+            });
 
             assert(
-              moduleConfig !== undefined,
-              `Unreachable. Module should not be undefined because moduleConfigsMap was created from serviceConfigs.modules. Searching for ${moduleUrlOrAbsolutePath} in ${JSON.stringify(
-                Object.fromEntries(
-                  moduleAbsolutePathOrURLToModuleConfigsMap.entries(),
-                ),
+              maybeServiceConfig !== undefined,
+              `Unreachable. Service should not be undefined because serviceConfigs where created from workerConfig.services. Looking for ${serviceName} in ${JSON.stringify(
+                serviceConfigs,
               )}`,
             );
 
-            const overridesFromProject = overrideModules?.[name];
+            const { overrideModules, serviceConfig } = maybeServiceConfig;
 
-            const overriddenModuleConfig = {
-              ...moduleConfig,
-              ...overridesFromService,
-              ...overridesFromProject,
-            };
+            const { [FACADE_MODULE_NAME]: facadeModule, ...restModules } =
+              serviceConfig.modules;
+
+            const modules = [
+              ...Object.entries(restModules),
+              [FACADE_MODULE_NAME, facadeModule] as const,
+            ].map(([name, { get, ...overridesFromService }]) => {
+              const moduleUrlOrAbsolutePath = getUrlOrAbsolutePath(
```

Comment:
maybe move that to a separate function? indentation level is pretty big here.

Reply:
Moved it to a separate function. Not sure if I made it a lot better